feature/s3/manager/download_test.go

```diff
@@ -577,6 +672,63 @@ func TestDownload_WithFailure(t *testing.T) {
 	}
 }
 
+func TestDownload_WithMismatch(t *testing.T) {
```

Comment:
took me a bit to read this, but when you make a call to S3 with an `IfMatch` that doesn't match, you get back a `PreconditionFailed` error.

Reply:
yea, this case just ensures downloader could react to the specific error